census/core.py

```diff
@@ -36,7 +36,42 @@ def list_or_str(v):
         return v
     return [v]
 
+def cast_nulls(func):
+    """
+    Decorator to format null values in API result casting functions.
+    """
+    def null_wrapper(v, cast_nulls):
+        # This value indicates that there were too few observations to compute
+        # an estimate. See:
+        # https://www.census.gov/data/developers/data-sets/acs-1year/notes-on-acs-estimate-and-annotation-values.html
+        if str(v) == '-666666666':
+            if cast_nulls is True:
+                return None
+            else:
+                raise NullValueException('Unhandled coded value: ', str(v))
+        else:
+            return func(v)
+    return null_wrapper
```

Comment:
Another (maybe simpler) way to do this would be to avoid the decorator abstraction and use a "checking" function instead. I liked the decorator since it felt to me like a semantic way of indicating that this checking logic is a high-level modification that can be applied somewhat abstractly to the other type casting functions.

Reply:
I think I'd be a bit more comfortable with this architecture if we had more cases than just the `-666666666`. That said I'm comfortable to go with this approach until we accumulate more cases, and then we can potentially refactor if appropriate.